libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,225 @@
+            }
+
+            response.StatusCode = (int)HttpStatusCode.NotFound;
+            logger.LogError($"Unknown verb and path: {request.Verb} {request.Path}");
+
+            return response;
+        }
+
+        /// <summary>
+        /// Build and return versioning information used for telemetry, including:
+        /// The Schema version is 3.1, put into the Microsoft-BotFramework header,
+        /// Protocol Extension Info,
+        /// The Client SDK Version
+        ///  https://github.com/Microsoft/botbuilder-dotnet/blob/d342cd66d159a023ac435aec0fdf791f93118f5f/doc/UserAgents.md,
+        /// Additional Info.
+        /// https://github.com/Microsoft/botbuilder-dotnet/blob/d342cd66d159a023ac435aec0fdf791f93118f5f/doc/UserAgents.md.
+        /// </summary>
+        /// <returns>A string containing versioning information.</returns>
+        private static string GetUserAgent() =>
+            string.Format(
+                "Microsoft-BotFramework/3.1 Streaming-Extensions/1.0 BotBuilder/{0} ({1}; {2}; {3})",
+                ConnectorClient.GetClientVersion(new ConnectorClient(new Uri("http://localhost"))),
+                ConnectorClient.GetASPNetVersion(),
+                ConnectorClient.GetOsVersion(),
+                ConnectorClient.GetArchitecture());
+
+        /// <summary>
+        /// Performs the actual processing of a request, handing it off to the adapter and returning the response.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <param name="response">The response to update and return, ultimately sent to client.</param>
+        /// <param name="logger">Optional logger.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The response ready to send to the client.</returns>
+        private async Task<StreamingResponse> ProcessStreamingRequestAsync(ReceiveRequest request, StreamingResponse response, ILogger<RequestHandler> logger, CancellationToken cancellationToken)
+        {
+            var body = string.Empty;
+
+            try
+            {
+                body = request.ReadBodyAsString();
+            }
+            catch (Exception ex)
+            {
+                response.StatusCode = (int)HttpStatusCode.BadRequest;
+                logger.LogError("Request body missing or malformed: " + ex.Message);
+
+                return response;
+            }
+
+            try
+            {
+                var adapter = new BotFrameworkStreamingExtensionsAdapter(_transportServer, _middlewareSet, logger);
+
+                // Check for dependency injected bot first, then check for bot passed as parameter, throw if neither is found.
+                var bot = _services?.GetService<IBot>() ?? this._bot ?? throw new Exception("Unable to find bot when processing request.");
```

Comment:
> var bot = _services?.GetService<IBot>() ?? this._bot ?? throw new Exception("Unable to find bot when processing request."); [](start = 15, length = 124)

switch to using if-else instead of ternary operators..this is hard to read